File: backend/internal/types/movies.go
package types

import (
	"time"

	"github.com/google/uuid"
)

type Movie struct {
	ID            uuid.UUID `json:"id"`
	ImdbID        string    `json:"imdb_id"`
	TmdbID        int       `json:"tmdb_id"`
	OriginalTitle string    `json:"original_title"`
	Title         string    `json:"title"`
	PosterPath    string    `json:"poster_path"`
	BackdropPath  string    `json:"backdrop_path"`
	Overview      string    `json:"overview"`
	ReleaseDate   string    `json:"release_date"`
	VoteAverage   float64   `json:"vote_average"`
	VoteCount     int       `json:"vote_count"`
	Revenue       int       `json:"revenue"`
	Homepage      string    `json:"homepage"`
	Genre         []Genre   `json:"genre"`
	Runtime       int       `json:"runtime"`
	Trailer       string    `json:"trailer"`
	Tagline       string    `json:"tagline"`
}

type UserMovieMeta struct {
	TmdbID        int32       `json:"tmdb_id"`
	MovieID       uuid.UUID   `json:"movie_id"`
	IsInWatchlist interface{} `json:"is_in_watchlist"`
	Diary         *UserDiary  `json:"diary,omitempty"`
	Review        *UserReview `json:"review,omitempty"`
}

type UserDiary struct {
	ID        uuid.UUID `json:"id"`
	WatchedAt time.Time `json:"watched_at"`
	CreatedAt time.Time `json:"created_at"`
}

type UserReview struct {
	ID        uuid.UUID `json:"id"`
	Text      string    `json:"text"`
	Rating    float64   `json:"rating"`
	IsSpoiler bool      `json:"is_spoiler"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

File: frontend/src/types/movies.ts
import type { Review } from "./reviews";

export type MovieDetails = {
  movie: MovieDetail;
  reviews: Review[];
  user_metadata: UserMovieMeta;
};

export type MovieDetail = {
  id: string;
  imdb_id: string;
  tmdb_id: number;
  original_title: string;
  title: string;
  poster_path: string;
  backdrop_path: string;
  overview: string;
  release_date: string;
  vote_average: number;
  vote_count: number;
  revenue: number;
  homepage: string;
  genre: Array<{
    id: number;
    name: string;
  }>;
  trailer: string;
  runtime: number;
  tagline: string;
};

export type Movie = {
  backdrop_path: string;
  genre_ids: Array<number>;
  genre: Array<string>;
  id: number;
  original_language: string;
  original_title: string;
  overview: string;
  poster_path: string;
  release_date: string;
  title: string;
  vote_average: number;
  vote_count: number;
};

export type UserMovieMeta = {
  movie_id: string;
  is_in_watchlist: boolean;
  tmdb_id: number;
  diary?: {
    id: string;
    watched_at: string;
    created_at: string;
  };

  review?: {
    id: string;
    text: string;
    rating: number;
    is_spoiler: boolean;
    created_at: string;
    updated_at: string; // fixed typo from updated_t
  };
};
